play-services-core/src/main/kotlin/com/google/android/gms/locationsharingreporter/service/LocationSharingReporterApiService.kt

```diff
@@ -0,0 +1,240 @@
+import com.google.android.gms.common.api.ApiMetadata
+import com.google.android.gms.common.api.CommonStatusCodes
+import com.google.android.gms.common.api.Status
+import com.google.android.gms.common.api.internal.IStatusCallback
+import com.google.android.gms.common.internal.ConnectionInfo
+import com.google.android.gms.common.internal.GetServiceRequest
+import com.google.android.gms.common.internal.IGmsCallbacks
+import com.google.android.gms.locationsharingreporter.LocationUploadRequest
+import com.google.android.gms.locationsharingreporter.NoticeAckedUpdateRequest
+import com.google.android.gms.locationsharingreporter.PeriodicLocationReportingIssues
+import com.google.android.gms.locationsharingreporter.PeriodicLocationUploadRequest
+import com.google.android.gms.locationsharingreporter.StartLocationReportingRequest
+import com.google.android.gms.locationsharingreporter.StopLocationReportingRequest
+import com.google.android.gms.locationsharingreporter.internal.ILocationReportingIssuesCallback
+import com.google.android.gms.locationsharingreporter.internal.ILocationReportingStatusCallbacks
+import com.google.android.gms.locationsharingreporter.internal.ILocationSharingReporterService
+import com.google.android.gms.locationsharingreporter.internal.ILocationUploadCallbacks
+import com.google.android.gms.locationsharingreporter.service.LocationSharingUpdate.Companion.startUpdateLocation
+import com.google.android.gms.locationsharingreporter.service.LocationSharingUpdate.Companion.stopUpdateLocation
+import com.google.android.gms.locationsharingreporter.service.ReportingRequestStoreFile.isLocationSharingEnabled
+import kotlinx.coroutines.Dispatchers
+import kotlinx.coroutines.launch
+import kotlinx.coroutines.withContext
+import org.microg.gms.BaseService
+import org.microg.gms.common.GmsService
+import org.microg.gms.common.PackageUtils
+import java.util.concurrent.ExecutionException
+
+private const val TAG = "LocationSharingReporter"
+val FEATURES = arrayOf(
+        Feature("LOCATION_SHARING_REPORTER", 2),
+        Feature("LOCATION_SHARING_REPORTER_SYNC", 1),
+        Feature("PERIODIC_LOCATION_SHARING_REPORTER", 1),
+        Feature("START_LOCATION_REPORTING", 1),
+        Feature("STOP_LOCATION_REPORTING", 1),
+        Feature("GET_REPORTING_ISSUES", 1),
+        Feature("UPDATE_NOTICE_STATE", 1),
+        Feature("LOCATION_SHARING", 1),
+)
+
+class LocationSharingReporterApiService : BaseService(TAG, GmsService.LOCATION_SHARING_REPORTER) {
+
+    override fun handleServiceRequest(callback: IGmsCallbacks, request: GetServiceRequest, service: GmsService?) {
+        Log.d(TAG, "handleServiceRequest: $request")
+        val packageName = PackageUtils.getAndCheckCallingPackage(this, request.packageName)
+                ?: throw IllegalArgumentException("Missing package name")
+        val callingPackageName = PackageUtils.getCallingPackage(this) ?: packageName
+        callback.onPostInitCompleteWithConnectionInfo(
```

Comment:
There is no access control check here yet. This means third-party apps can connect to this API without authorization or user consent.